spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/transaction/jta/NarayanaJtaConfiguration.java

```diff
@@ -114,10 +115,12 @@ public TransactionManager narayanaTransactionManager() {
 	@Bean
 	@DependsOn("narayanaConfiguration")
 	public RecoveryManagerService narayanaRecoveryManagerService() {
+		RecoveryManager.delayRecoveryManagerThread();
```

Comment:
Given that this is a static method, what are the implications for integration tests? Will this still work if transaction recovery runs multiple times in the same JVM?